cpp/include/cudf/io/types.hpp

```diff
@@ -99,6 +99,20 @@ enum statistics_freq {
   STATISTICS_COLUMN   = 3,  ///< Full column and offset indices. Implies STATISTICS_ROWGROUP
 };
 
+/**
+ * @brief Valid parquet encodings for use with `column_in_metadata::set_encoding()`
+ */
+struct parquet_encoding {
```

Comment:
Note on my thinking here (or lack thereof)...I was thinking using strings to specify the desired encoding would be better than an enum since the `column_input_metadata` is shared between multiple encoders, and it would be more natural to use strings with a CLI or through the python interface. And if ORC has different encoding names, then we could add another set of constants for that.

 But a user interface could translate string values to an enum, and the enum could just add as many fields as necessary, some not relevant to one implementation or the other, so maybe this is silly. This acts like a scoped enum already, so I'm not opposed to switching.

Reply:
It does seem like this could be an enum :) I don't think a lot of code would change.

Reply:
Actually I rushed a bit with this comment. Would this become just `encoding` and contain a superset of all encoding types?

Reply:
Yeah, probably, or `page_encoding` maybe. But it would have to apply to all encoders that use the metadata (which I assume is just parquet and orc for now).

Reply:
`page_encoding` isn't a great name for ORC...how about `column_encoding`?

Reply:
Can't believe I forgot my boy ORC.
sounds good!